src/main/java/ch/njol/skript/lang/ExpressionSection.java

```diff
@@ -0,0 +1,99 @@
+package ch.njol.skript.lang;
+
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.expressions.base.SectionExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * A dummy trigger item representing the 'section' aspect of a {@link SectionExpression}.
+ * This is not a parsed or registered syntax in itself, and can be used for getting access to the parse-time features
+ * of a section syntax (e.g. loading body code).
+ * <p>
+ * This is not safe to be used during runtime, since it is not a part of the trigger item tree.
+ *
+ * @see SectionExpression
+ * @see Section
+ * @see EffectSectionEffect
+ */
+@ApiStatus.Internal
+public class ExpressionSection extends Section {
+
+	protected final SectionExpression<?> expression;
+
+	public ExpressionSection(SectionExpression<?> expression) {
+		this.expression = expression;
+	}
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		SectionContext context = getParser().getData(SectionContext.class);
+		return this.init(expressions, matchedPattern, isDelayed, parseResult, context.sectionNode, context.triggerItems)
+			&& context.claim(expression);
+	}
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed,
+						SkriptParser.ParseResult parseResult,
+						@Nullable SectionNode sectionNode, List<TriggerItem> triggerItems) {
+		return expression.init(expressions, matchedPattern, isDelayed, parseResult, sectionNode, triggerItems);
+	}
+
+	@Override
+	protected @Nullable TriggerItem walk(Event event) {
+		return super.walk(event, false);
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return expression.toString(event, debug);
+	}
+
+	public SectionExpression<?> getAsExpression() {
+		return expression;
+	}
+
+	@Override
+	public void loadCode(SectionNode sectionNode) {
```

Comment:
Are these overridden just to update the access modifiers? I think it might be better to consider putting this class and `SectionExpression` in the same package. In fact, might this be fine to go in the same file as SectionExpression? It does not seem like it needs to be exposed.